### PR TITLE
Add csrf protection

### PR DESCRIPTION
**Pull request by seallard**
## Description
Since some of our endpoints only rely on cookies for authentication they are vulnerable to CSRF. An attacker could force a user to run arbitrary requests, for instance deleting resources (invoices, samples, customers etc.).

### Fixed
- CSRF vulnerability for the cg API (the admin and invoice endpoints).

### How to prepare for test
Use the Github action for deploying this branch to the cg-vm1 stage environment.

### How to test

- [x] Open https://cg-statusdb-stage.scilifelab.se/ and sign in.
- [x] Make sure that the order form can be filled in as usual via the order interface.
- [x] Make sure that records can be updated with legitimate requests via the admin interface.
- [x] Make sure that invoices can be updated with legitimate requests via the invoice interface.
- [x] Try to run a CSRF exploit for an admin endpoint deleting some resource.
- [x] Try to run a CSRF exploit  for an invoice endpoint changing some resource.

### Expected test outcome
- [x] Check that a `Bad Request` error is given when the CSRF exploit is run.
- [x] Take a screenshot and attach or copy/paste the output.

## Review

- [x] Tests executed by SA
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Deploy this branch on prod.

### CODE REVIEW

**Comment by seallard** [comment]
image

File: cg/server/invoices/views.py [hunk 1]
```diff
@@ -143,12 +147,9 @@ def new(record_type):
     )
 
 
-@BLUEPRINT.route("/<int:invoice_id>", methods=["GET", "POST"])
```
Comment (seallard) [inline cg/server/invoices/views.py:146]: I could not find any call to this endpoint with POST.

**Review by Vince-janv — APPROVED** [review]
This is great 🤩 Did you also test that excepting the api works as intended?

**Comment by seallard** [comment]
Deployed to stage and prod.

**Pull request merged**